components/NavigationBar/tests/unit/NavigationBarButtonTitleFontTests.swift

```diff
@@ -0,0 +1,126 @@
+
+  override func setUp() {
+    navigationBar = MDCNavigationBar()
+  }
+
+  private func recursiveSubviews(of superview: UIView) -> [UIView] {
+    var subviews = superview.subviews
+    for subview in superview.subviews {
+      subviews.append(contentsOf: recursiveSubviews(of: subview))
+    }
+    return subviews
+  }
+
+  func testDefaultFontBehavior() {
+    // Given
+    navigationBar.leadingBarButtonItems = [
+      UIBarButtonItem(title: "Text", style: .plain, target: nil, action: nil)
+    ]
+    navigationBar.trailingBarButtonItems = [
+      UIBarButtonItem(title: "Text 2", style: .plain, target: nil, action: nil)
+    ]
+
+    // Then
+    for view in recursiveSubviews(of: navigationBar) {
+      if let button = view as? MDCButton {
+        XCTAssertNil(button.titleFont(for: .normal))
+      }
+    }
+  }
+
+  func testCustomFontIsSetForNewButtons() {
+    // Given
+    let font = UIFont.systemFont(ofSize: 100)
+
+    // When
+    navigationBar.setButtonsTitleFont(font, for: .normal)
+    navigationBar.leadingBarButtonItems = [
+      UIBarButtonItem(title: "Text", style: .plain, target: nil, action: nil)
+    ]
+    navigationBar.trailingBarButtonItems = [
+      UIBarButtonItem(title: "Text 2", style: .plain, target: nil, action: nil)
+    ]
+
+    // Then
+    for view in recursiveSubviews(of: navigationBar) {
+      if let button = view as? MDCButton {
+        XCTAssertEqual(button.titleFont(for: .normal), font)
+        XCTAssertEqual(button.titleLabel?.font, font)
+      }
+    }
+  }
+
+  func testCustomFontIsSetForExistingButtons() {
+    // Given
+    let font = UIFont.systemFont(ofSize: 100)
+
+    // When
+    navigationBar.leadingBarButtonItems = [
+      UIBarButtonItem(title: "Text", style: .plain, target: nil, action: nil)
+    ]
+    navigationBar.trailingBarButtonItems = [
+      UIBarButtonItem(title: "Text 2", style: .plain, target: nil, action: nil)
+    ]
+    navigationBar.setButtonsTitleFont(font, for: .normal)
+
+    // Then
+    for view in recursiveSubviews(of: navigationBar) {
+      if let button = view as? MDCButton {
+        XCTAssertEqual(button.titleFont(for: .normal), font)
+        XCTAssertEqual(button.titleLabel?.font, font)
+      }
+    }
+  }
+
+  func testCustomFontFallbackBehavior() {
+    // Given
+    let normalFont = UIFont.systemFont(ofSize: 100)
+    let selectedFont = UIFont.systemFont(ofSize: 50)
+
+    // When
+    navigationBar.setButtonsTitleFont(normalFont, for: .normal)
+    navigationBar.setButtonsTitleFont(selectedFont, for: .selected)
+    navigationBar.leadingBarButtonItems = [
+      UIBarButtonItem(title: "Text", style: .plain, target: nil, action: nil)
+    ]
+    navigationBar.trailingBarButtonItems = [
+      UIBarButtonItem(title: "Text 2", style: .plain, target: nil, action: nil)
+    ]
+
+    // Then
+    for view in recursiveSubviews(of: navigationBar) {
+      if let button = view as? MDCButton {
+        button.isSelected = true
```

Comment:
Disabled state? Highlighted? Highlighted + Selected?

Reply:
Filed https://www.pivotaltracker.com/story/show/156932122 to track this.